### PR TITLE
compareTo realization for Vector2 and Vector3

### PR DESCRIPTION
**Pull request by nikitamiroshnichenko**
Comparable java interface implementation for Vector2 and Vector3 classes. May be useful when you need to sort array of vectors.

### CODE REVIEW

File: gdx/src/com/badlogic/gdx/math/Vector2.java [hunk 1]
```diff
@@ -551,4 +551,24 @@ public Vector2 setZero () {
 		this.y = 0;
 		return this;
 	}
+
+	/** Compares this vector with the other {@link Vector2} based on x and y variables.
+     * Useful for sorting.
```
Comment (SimonIT) [inline gdx/src/com/badlogic/gdx/math/Vector2.java]: Please fix the indentation

Comment (nikitamiroshnichenko) [reply gdx/src/com/badlogic/gdx/math/Vector2.java]: done

**Comment by mgsx-dev** [comment]
Vector2 and Vector3 doesn't have natural ordering. It doesn't make sense to implement this. See https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html contract.

You can always use comparators based on specific ordering logic in your own code.

Please reconsider these changes.

**Comment by Darkyenus** [comment]
I agree with @mgsx-dev. While it sometimes might be useful to sort vectors, the required ordering is usually wildly different from case to case. Implementing `Comparable` would probably just be confusing and bug prone.

Additionally, if the goal of this PR is to provide *any* total ordering (which is probably the most unbiased/useful way to go about this, and it is what I would from javadoc expect), this PR doesn't provide that (logic is more complicated than it needs to be and `Vector3` doesn't consider all components).

Nevertheless, thanks for the PR, don't let this discourage you.

**Pull request closed without merging**